### ISSUE TITLE
How to get started with with TiKV development?

### ISSUE THREAD

**Issue by amirouche** [tikv/tikv#2416]
Héllo,

I'd like to get into TiKV development. This question kind of silly, I should just start reading the code. But I'd like to have a tip or two on how to get started. I have prior knowledge of rocksdb, so maybe looking at that part of the code will be easier for me. I am already following the project and reading some PRs and issues. I already went through [kvproto](https://github.com/pingcap/kvproto) but I have an hard time relating TiKV gRPC with raw rocksdb API. For instance, how do you start a transaction?

Basically, how do you recommend approaching the code base?

TIA

**Comment by amirouche** [comment]
Can you recommend some readings, tutorials, courses to better grasp TiKV inner working?

**Comment by siddontang** [comment]
Thanks @amirouche 

You can browse our English blog https://pingcap.github.io/blog/ 

I have to say that TiKV misses documents and tutorials now, which makes the user hard to contribute. We have been improving this 😭 

We implement our transaction from by Google Percolator. You can see the paper at first, then browse the tikv client codes in TiDB. Maybe this [article](https://pingcap.github.io/blog/2016/11/17/mvcc-in-tikv/) can help you too.

It is very appreciated that you can contribute to TiKV. If you have any problem, please tell us.

**Comment by Hoverbear** [comment]
Hi @amirouche !

Why don't you get in touch with me on Slack? https://join.slack.com/t/tikv-wg/shared_invite/enQtNTUyODE4ODU2MzI0LTgzZDQ3NzZlNDkzMGIyYjU1MTA0NzIwMjFjODFiZjA0YjFmYmQyOTZiNzNkNzg1N2U1MDdlZTIxNTU5NWNhNjk

I'd be happy to help you or anyone get started. You'll also see we know have things like the https://github.com/tikv/client-rust to help you play with the codebase.
